bundles/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBConfiguration.java

```diff
@@ -0,0 +1,190 @@
+    private String url;
+    private String user;
+    private String password;
+    private String token;
+    private String databaseName;
+    private String retentionPolicy;
+    private InfluxDBVersion version;
+
+    private boolean replaceUnderscore;
+    private boolean addCategoryTag;
+    private boolean addTypeTag;
+    private boolean addLabelTag;
+
+    @SuppressWarnings("null")
+    public InfluxDBConfiguration(Map<String, @Nullable Object> config) {
+        url = (String) config.getOrDefault(URL_PARAM, "http://127.0.0.1:8086");
+        user = (String) config.getOrDefault(USER_PARAM, "openhab");
+        password = (String) config.getOrDefault(PASSWORD_PARAM, "");
+        token = (String) config.getOrDefault(TOKEN_PARAM, "");
+        databaseName = (String) config.getOrDefault(DATABASE_PARAM, "openhab");
+        retentionPolicy = (String) config.getOrDefault(RETENTION_POLICY_PARAM, "autogen");
+        version = parseInfluxVersion(config.getOrDefault(VERSION_PARAM, InfluxDBVersion.V1.name()));
+
+        replaceUnderscore = getConfigBooleanValue(config, REPLACE_UNDERSCORE_PARAM, false);
+        addCategoryTag = getConfigBooleanValue(config, ADD_CATEGORY_TAG_PARAM, false);
+        addLabelTag = getConfigBooleanValue(config, ADD_LABEL_TAG_PARAM, false);
+        addTypeTag = getConfigBooleanValue(config, ADD_TYPE_TAG_PARAM, false);
+    }
+
+    private static boolean getConfigBooleanValue(Map<String, @Nullable Object> config, String key,
+            boolean defaultValue) {
+        Object object = config.get(key);
+
+        if (object instanceof Boolean) {
+            return (Boolean) object;
+        } else if (object != null) {
+            return "true".equalsIgnoreCase((String) object);
+        } else {
+            return defaultValue;
+        }
+    }
+
+    private InfluxDBVersion parseInfluxVersion(@Nullable Object value) {
+        try {
+            return InfluxDBVersion.valueOf((String) value);
+        } catch (RuntimeException e) {
+            logger.warn("Invalid version {}", value);
+            return InfluxDBVersion.UNKNOWN;
+        }
+    }
+
+    public boolean isValid() {
+        boolean hasVersion = version != InfluxDBVersion.UNKNOWN;
+        boolean hasCredentials = false;
+        if (version == InfluxDBVersion.V1) {
+            hasCredentials = StringUtils.isNotBlank(user) && StringUtils.isNotBlank(password);
+        } else if (version == InfluxDBVersion.V2) {
+            hasCredentials = StringUtils.isNotBlank(token)
+                    || (StringUtils.isNotBlank(user) && StringUtils.isNotBlank(password));
+        }
+        boolean hasDatabase = StringUtils.isNotBlank(databaseName);
+        boolean hasRetentionPolicy = StringUtils.isNotBlank(retentionPolicy);
+
+        boolean valid = hasVersion && hasCredentials && hasDatabase && hasRetentionPolicy;
+        if (valid) {
+            return true;
+        } else {
```

Comment:
The if block already returns, so using an else here is redundant. Also, less indentation looks better 😄

Reply:
This one breaks my heart,  I really don't like non-symmetric ifs.
But no problem adapting that if this is codebase style.

Reply:
Keep it as is then, I don't want to break your heart. 😉